packages/cli-kit/src/environment/service.test.ts

```diff
@@ -1,6 +1,12 @@
 import {isSpinEnvironment, serviceEnvironment} from './service.js'
 import {Environment} from '../network/service.js'
-import {expect, it, describe} from 'vitest'
+import {expect, it, describe, afterEach} from 'vitest'
+
+const OLD_ENV = {...process.env}
+
+afterEach(() => {
+  process.env = {...OLD_ENV}
+})
```

Comment:
ah, missed that part of the file, sorry!

But i think a better solution would be to pass the custom env as a paremeter, like we do in other tests:

```ts
  it('returns false when not working with spin instances', () => {
    // Given
    const env = {SHOPIFY_SERVICE_ENV: 'local'}

    // When
    const got = isSpinEnvironment(env)

    // Then
    expect(got).toBe(false)
  })
```

This is the reason why `isSpinEnvironment` accepts a custom env, to override it in testing :)

Reply:
yes, in this case that option is available, but I think this could be a solution for other places where that param is not available, and even for testing the method without using the param which is optional. 
I can change that here but I thought It could be a `reference` for a future needing of overwriting some env variable

Reply:
Not sure about that, in my opinion tests should never change `process.env` so there shouldn't be a need for restoring it, and if a function depends on it, it should be injectable.

Reply:
ok, do you think we need any kind of lint rule or so to avoid set `process.env` in test files?

Reply:
I agree with @isaacroldan here. If the subject under test interacts with a global variable like `process.env` we should refactor the implementation to get the variable injected. Mocking a global state might yield unexpected behaviours when running tests.